### PR TITLE
Better Short Pages

### PR DESCRIPTION
**Pull request by Ap0c**
## Why are you doing this?

This builds on the work done in #103, #105 and #115... turns out getting a sticky footer right is hard, @svillafe and I have tried multiple iterations over the last few weeks.

Anyway, our current solution was a bit wonky, and I immediately ran into a problem with it working on the contributions landing page. We were flexing a lot of things, but it occurred to me that we only really need to flex the last section on the page to push the footer down. So say hello to `gu-content-filler`, a class that can be applied to the last section to do just this.

The benefit of this solution is that we don't have to hack this functionality into `gu-content-margin`, it works independently and need only be applied on pages that are too short for the viewport. It also solves the issue with the previous implementation where content was being pushed off the bottom of the page even in tall viewports, resulting in lots of whitespace and unneeded scrollbars.

Currently I've applied empty filler divs to most of the pages where we need this, to avoid messing with the codebase too much. However, I'd like to do a follow up PR where the sections on these pages get refactored (because they're a bit wonky at the moment), allowing the class to be applied to an existing section with content. It might be worth converting `gu-content`, `gu-content-margin`, `gu-header-margin` and `gu-content-filler` to mixins at this point as well (see [this trello card](https://trello.com/c/xJzW2QC6/728-section-convention-and-mixins)).

## Changes

- Removed flexing from `gu-content-margin` and added a `gu-content-filler` class to handle it instead (this is in `layout.scss`, it's quite a simple change but the diff has come out weird).
- Removed additional `gu-content` classes and applied filler divs instead to a number of pages.

## Screenshots

I'm just going to show the one page because it's the same across all of them...

**Thank You Page - Before:**

This has a scrollbar.

footer-previous

**Thank You Page - After:**

This does not.

footer-filler

### CODE REVIEW

File: assets/pages/contributions-thankyou/oneoffContributionsThankyou.jsx [hunk 1]
```diff
@@ -59,6 +59,9 @@ const content = (
         </InfoSection>
       </div>
     </section>
+    <section className="gu-content-filler">
```
Comment (davidfurey) [inline assets/pages/contributions-thankyou/oneoffContributionsThankyou.jsx]: If this is going to be necessary on all pages, can we avoid the duplication by putting it in the SimpleFooter template?

File: assets/pages/monthly-contributions/monthlyContributions.scss [hunk 2]
```diff
@@ -106,4 +106,14 @@
 		}
 	}
 
+	// ----- Filler
+
+	.gu-content-filler {
```
Comment (svillafe) [inline assets/pages/monthly-contributions/monthlyContributions.scss]: This piece of CSS is spread across three pages, Is it possible to create a shared component that receives the colour in a prop?

**Comment by Ap0c** [comment]
Figured out a way of doing this without filler divs @davidfurey, and dropping the additional CSS @svillafe.

**Comment by svillafe** [comment]
👍 Nice that you deleted all that duplication 🎉

**Comment by davidfurey** [comment]
👍 🔨

**Comment by Ap0c** [comment]
Working on PROD.

**Pull request merged**